src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -11599,11 +11606,11 @@ void Assembler::vex_prefix(Address adr, int nds_enc, int xreg_enc, VexSimdPrefix
             (attributes->get_vector_len() == AVX_512bit) ||
             (!_legacy_mode_vl) ||
             (attributes->is_legacy_mode())),"XMM register should be 0-15");
-    assert(((nds_enc < 16 && xreg_enc < 16) || (!attributes->is_legacy_mode())),"XMM register should be 0-15");
+    assert(((nds_enc < 16 && xreg_enc < 16) || has_gpr || (!attributes->is_legacy_mode())),"XMM register should be 0-15");
```

Comment:
This could be:
assert((!is_extended || (!attributes->is_legacy_mode())),"XMM register should be 0-15");

Reply:
Thanks, done.